mitosheet/src/mito/utils/actions.tsx

```diff
@@ -1649,6 +1649,106 @@ export const getActions = (
             searchTerms: ['docs', 'documentation', 'help', 'support'],
             tooltip: "Documentation, tutorials, and how-tos on all functionality in Mito."
         },
+        [ActionEnum.Select_Columns]: {
+            type: 'build-time',
+            staticType: ActionEnum.Select_Columns,
+            longTitle: 'Select column',
+            actionFunction: () => {
+                // We turn off editing mode, if it is on
+                setEditorState(undefined);
+
+                // We close the editing taskpane if its open
+                closeOpenEditingPopups();
+
+                // We open the column selection taskpane
+                let newStartingColumnIndex = gridState.selections[0].startingColumnIndex;
+                let newEndingColumnIndex = gridState.selections[0].endingColumnIndex;
+                for (const selection of gridState.selections) {
+                    newStartingColumnIndex = Math.min(newStartingColumnIndex, selection.startingColumnIndex);
+                    newEndingColumnIndex = Math.max(newEndingColumnIndex, selection.endingColumnIndex);
+                }
+                setGridState(prevGridState => {
+                    return {
+                        ...prevGridState,
+                        selections: [{
+                            startingColumnIndex: newStartingColumnIndex ?? 0,
+                            endingColumnIndex: newEndingColumnIndex ?? 0,
+                            startingRowIndex: -1,
+                            endingRowIndex: -1,
+                            sheetIndex: sheetIndex
+                        }]
+                    }
+                });
+            },
+            isDisabled: () => {return doesAnySheetExist(sheetDataArray) ? defaultActionDisabledMessage : 'There are no columns to select. Import data.'},
+            searchTerms: ['select', 'columns', 'select columns'],
+            tooltip: "Select columns for all cells in currently selected range."
+        },
+        [ActionEnum.Select_Rows]: {
+            type: 'build-time',
+            staticType: ActionEnum.Select_Rows,
+            longTitle: 'Select row',
+            actionFunction: () => {
+                // We turn off editing mode, if it is on
+                setEditorState(undefined);
+
+                // We close the editing taskpane if its open
+                closeOpenEditingPopups();
+
+                // We open the column selection taskpane
+                let newStartingRowIndex = gridState.selections[0].startingRowIndex;
+                let newEndingRowIndex = gridState.selections[0].endingRowIndex;
+                for (const selection of gridState.selections) {
+                    console.log(selection)
+                    newStartingRowIndex = Math.min(newStartingRowIndex, selection.startingRowIndex);
+                    newEndingRowIndex = Math.max(newEndingRowIndex, selection.endingRowIndex);
```

Comment:
Wanna try the same as above the other action here (e.g. just working with the first selection), and see if it fixes things?